core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -279,8 +279,10 @@ object DumpLogSegments {
 
               if (batch.magic >= RecordBatch.MAGIC_VALUE_V2) {
                 print(" sequence: " + record.sequence + " headerKeys: " + record.headers.map(_.key).mkString("[", ",", "]"))
-              } else {
-                print(s" crc: ${record.checksumOrNull} isvalid: ${record.isValid}")
+              }
+              record match {
+                case r: AbstractLegacyRecordBatch => print(s" crc: ${r.checksum}}")
```

Comment:
Shall we include the `isValid` part here too?

Reply:
sure!